docs/rules/prefer-find-by.md

```diff
@@ -0,0 +1,43 @@
+# Use `find*` query methods to wait for elements instead of waitFor (prefer-find-by)
+
+TBD
+
+## Rule details
+
+This rule aims to use `findBy*` or `findAllBy*` queries to wait for elements, rather than using `waitFor`, or the deprecated methods `waitForElement` and `wait`
+
+Examples of **incorrect** code for this rule
+
+```js
+const submitButton = await waitFor(() =>
+  screen.getByRole('button', { name: /submit/i })
+);
+
+const submitButton = await waitFor(() =>
+  screen.getAllTestId('button', { name: /submit/i })
+);
+
+const submitButton = await waitFor(() =>
+  queryByLabel('button', { name: /submit/i })
+);
+
+const submitButton = await waitFor(() =>
+  queryAllByText('button', { name: /submit/i })
+);
+```
+
+Examples of **correct** code for this rule:
+
+```js
+const submitButton = await findByText('foo');
+
+const submitButton = await screen.findAllByRole('table');
+```
+
+## When Not To Use It
+
+TBD
```

Comment:
```suggestion
Don't use those rule if you don't care if people aren't following best practices of testing library.
```

Reply:
I wouldn't go that _direct_ here. Better something nicer like "Not encouraging use of `findBy` shortcut from testing library best practices".